iocore/net/SSLNetVConnection.cc

```diff
@@ -207,11 +207,17 @@ ssl_read_from_net(SSLNetVConnection *sslvc, EThread *lthread, int64_t &ret)
   while (sslErr == SSL_ERROR_NONE) {
     int64_t block_write_avail = buf.writer()->block_write_avail();
     if (block_write_avail <= 0) {
-      buf.writer()->add_block();
-      block_write_avail = buf.writer()->block_write_avail();
-      if (block_write_avail <= 0) {
-        Warning("Cannot add new block");
+      // If we filled up one block, give back to the event loop so we don't
+      // overbuffer.
+      if (bytes_read > 0) {
```

Comment:
Ok, I will replace my logic that is going per block and explicitly adding blocks and call buf.writer()->writer_avail() instead and rely on that to do the appropriate water mark calculations and add new blocks as necessary.